### PR TITLE
Type annotation effort 2019-04-03

### PR DESCRIPTION
**Pull request by pgunn**
This:
* adds more type annotations
* corrects the spelling of the word 'indices' across the codebase
* Avoids use of 'dict' as a variable name (as that's also a type, so it's confusing)
* removes a few functions that were not being used anywhere (and other cruft)
* has some very minor refactoring
* Adds some TODOs to think about for later work
* makes docs a little better in some places

### CODE REVIEW

File: caiman/cluster.py [hunk 1]
```diff
@@ -34,27 +34,13 @@
 import subprocess
 import sys
 import time
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 from .mmapping import load_memmap
 
 logger = logging.getLogger(__name__)
 
-def get_patches_from_image(img, shapes, overlaps):
```
Comment (epnev) [inline caiman/cluster.py]: @pgunn Why is this method being removed?

File: caiman/components_evaluation.py [hunk 2]
```diff
@@ -29,52 +29,14 @@
 except:
     pass
 
-def estimate_noise_mode(traces, robust_std=False, use_mode_fast=False, return_all=False):
```
Comment (epnev) [inline caiman/components_evaluation.py]: @pgunn This is also being removed?

File: caiman/components_evaluation.py [hunk 3]
```diff
@@ -291,8 +249,8 @@ def evaluate_components_CNN(A, dims, gSig, model_name=os.path.join(caiman_datadi
 #%%
 
 
-def evaluate_components(Y, traces, A, C, b, f, final_frate, remove_baseline=True, N=5, robust_std=False,
-                        Athresh=0.1, Npeaks=5, thresh_C=0.3, sigma_factor=3.):
+def evaluate_components(Y:np.ndarray, traces:np.ndarray, A, C, b, f, final_frate, remove_baseline:bool=True, N:int=5, robust_std:bool=False,
```
Comment (epnev) [inline caiman/components_evaluation.py]: @pgunn I think that Y can also be a memory mapped array here.

File: caiman/cluster.py (hunk 1 above)
Comment (pgunn) [reply caiman/cluster.py]: Was unused. If that's not a reason to remove it I can revert this part. (same as below)

**Comment by epnev** [comment]
@pgunn Can you move the unused function into a separate file? Perhaps start an unused.py file in the utils folder. I can see both of these functions being useful at some point, so wouldn't want to completely get rid of them at this point. Once you do that and do a further rebase with the current dev I can merge.

**Comment by pgunn** [comment]
> @pgunn Can you move the unused function into a separate file? Perhaps start an unused.py file in the utils folder. I can see both of these functions being useful at some point, so wouldn't want to completely get rid of them at this point. Once you do that and do a further rebase with the current dev I can merge.

I made an unused.py as requested.

**Pull request merged**